### ISSUE TITLE
AIP-231: plural method naming does not account for resource.plural

### ISSUE THREAD

**Issue by loeffel-io** [googleapis/api-linter#1566]
Please correct me if i am wrong

- [x] the plural of sku should be skus?
- [x] the batchGet linter message is incorrect i think?

```proto
  rpc BatchGetSkus(BatchGetSkusRequest) returns (BatchGetSkusResponse) {
    option (google.api.http) = {
      get: "/v1/{parent=projects/*/services/*}/skus:batchGet"
    };
    option (google.api.method_signature) = "parent,names";
  }

// Represents the batch get sku response
message BatchGetSkusResponse {
  // The skus
  repeated Sku skus = 1;
}

// Represents the sku resource
message Sku {
  option (google.api.resource) = {
    type: "billing.mindful.com/Sku",
    pattern: "projects/{project}/services/{service}/skus/{sku}",
    singular: "sku",
    plural: "skus"
  };
  // The resource name
  // Format: projects/{project}/services/{service}/skus/{sku}
...
```

```yaml
- file_path: mindful/earth/billing/v1/sku.proto
  problems:
    - message: The resource part in method "BatchGetSkus" should not be "Skus", but should be its plural form "Skuses"
      location:
        start_position:
            line_number: 40
            column_number: 7
        end_position:
            line_number: 40
            column_number: 18
        path: mindful/earth/billing/v1/sku.proto
      rule_id: core::0231::plural-method-name
      rule_doc_uri: https://linter.aip.dev/231/plural-method-name
    - message: Message "BatchGetSkusResponse" has no "Skus" type field
      location:
        start_position:
            line_number: 209
            column_number: 1
        end_position:
            line_number: 212
            column_number: 1
        path: mindful/earth/billing/v1/sku.proto
      rule_id: core::0231::response-resource-field
      rule_doc_uri: https://linter.aip.dev/231/response-resource-field
2025/11/15 08:01:06 found problems during linting
```

**Comment by noahdietz** [comment]
Hey @loeffel-io, you are correct on both counts! I checked your boxes :)

We need to change the rule code here:
https://github.com/googleapis/api-linter/blob/2c8f81a6b6bbb8a7449a23d6219034d4439b987e/rules/aip0231/plural_method_name.go#L32-L37

to be `(google.api.resource).plural` aware (as per AIP-123) and fallback on the existing pluralization logic when that annotation is not present. This would likely be "discovered" by looking for the field in the response that is of type `message` and that `message` has `google.api.resource`.

Feel free to send a PR if you have a moment to do so, otherwise we will get to this when we can.

cc @quirogas

**Comment by noahdietz** [comment]
I took a crack at it and updated some of the other batch methods to fix them as well.

**Comment by loeffel-io** [comment]
thanks @noahdietz 🫶

**Comment by noahdietz** [comment]
Closing this as the fix should appear in v2.1.0.
